src/laddie/laddie_utilities.f90

```diff
@@ -55,6 +57,101 @@ SUBROUTINE compute_ambient_TS( mesh, laddie, ocean, ice)
 
   END SUBROUTINE compute_ambient_TS
 
+  SUBROUTINE map_H_a_b( mesh, laddie, H_a, H_b)
+    ! Map layer thickness from a to b grid, accounting for BCs
+
+    ! In- and output variables
+
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_laddie_model),                INTENT(IN)    :: laddie
+    REAL(dp), DIMENSION(mesh%vi1:mesh%vi2), INTENT(IN)    :: H_a
+    REAL(dp), DIMENSION(mesh%ti1:mesh%ti2), INTENT(INOUT) :: H_b
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'map_H_a_b'
+    INTEGER                                               :: i, vi, ti, n
+    REAL(dp), DIMENSION(mesh%nV)                          :: H_a_tot
+ 
+    ! Add routine to path
+    CALL init_routine( routine_name)
+
+    CALL gather_to_all_dp_1D( H_a, H_a_tot)
+
+    ! Get T and S at layer base
+    DO ti = mesh%ti1, mesh%ti2
+       H_b( ti) = 0.0_dp
+       IF (laddie%mask_b( ti)) THEN
+         
+         ! Set zero
+         n = 0
+
+         ! Loop over vertices
+         DO i = 1, 3
+           vi = mesh%Tri( ti, i)
+           IF (laddie%mask_a( vi)) THEN
+             H_b( ti) = H_b( ti) + H_a_tot( vi)
+             n = n + 1
+           END IF
+         END DO
+
+         H_b( ti) = H_b( ti) / n
+
+       END IF
+    END DO
+
+    ! Finalise routine path
+    CALL finalise_routine( routine_name)
+
+  END SUBROUTINE map_H_a_b
+
+  SUBROUTINE map_H_a_c( mesh, laddie, H_a, H_c)
+    ! Map layer thickness from a to c grid, accounting for BCs
+
+    ! In- and output variables
+
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_laddie_model),                INTENT(IN)    :: laddie
+    REAL(dp), DIMENSION(mesh%vi1:mesh%vi2), INTENT(IN)    :: H_a
+    REAL(dp), DIMENSION(mesh%ei1:mesh%ei2), INTENT(INOUT) :: H_c
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'map_H_a_c'
+    INTEGER                                               :: i, vi, ei, n
+    REAL(dp), DIMENSION(mesh%nV)                          :: H_a_tot
+ 
+    ! Add routine to path
+    CALL init_routine( routine_name)
+
+
+    CALL gather_to_all_dp_1D( H_a, H_a_tot)
+
+    DO ei = mesh%ei1, mesh%ei2
+       H_c( ei) = 0.0_dp
+       ! Set zero
+       n = 0
+       
+       ! Loop over vertices
```

Comment:
An edge can only ever have two vertices. Maybe cleaner to remove the loop altogether and just use vi, vj directly.